bciers/apps/reporting/src/app/components/facility/FacilityEmissionAllocationForm.tsx

```diff
@@ -145,13 +158,41 @@ export default function FacilityEmissionAllocationForm({
       products: initialData.report_product_emission_allocation_totals,
     },
   }));
+  const [shouldReset, setShouldReset] = useState(false);
 
   // State for submit button disable
   const [errors, setErrors] = useState<string[] | undefined>();
   const [submitButtonDisabled, setSubmitButtonDisabled] = useState(true);
 
   // 🔄 Check for allocation mismatch on page load to prevent submit
   useEffect(() => {
+    // If methodology is not applicable the products are removed, reset them here
+    if (formData.allocation_methodology !== "Not Applicable" && shouldReset) {
+      setFormData({
+        ...formData,
+        basic_emission_allocation_data:
+          initialData.report_product_emission_allocations
+            .filter((category: any) => category.category_type === "basic")
+            .map(calculateEmissionData),
+        fuel_excluded_emission_allocation_data:
+          initialData.report_product_emission_allocations
+            .filter(
+              (category: any) => category.category_type === "fuel_excluded",
+            )
+            .map(calculateEmissionData),
+        total_emission_allocations: {
+          facility_total_emissions: initialData.facility_total_emissions,
+          products: initialData.report_product_emission_allocation_totals,
+        },
+      });
+      setShouldReset(false);
+    }
+
+    if (formData.allocation_methodology === "Not Applicable") {
+      removeProducts(formData);
+      setShouldReset(true);
+    }
+
```

Comment:
The reset logic is just so that we don't submit information / show the right things to the user. I think we can make this better either with RJSF, or with a more succinct `useEffect` just on the allocation methodology selection.